solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -414,7 +416,19 @@ private IndexDeletionPolicyWrapper initDeletionPolicy(IndexDeletionPolicyWrapper
     } else {
       delPolicy = new SolrDeletionPolicy();
     }
-    return new IndexDeletionPolicyWrapper(delPolicy);
+
+    return new IndexDeletionPolicyWrapper(delPolicy, snapshotMgr);
+  }
+
+  private SolrSnapshotMetaDataManager initSnapshotMetaDataManager() {
```

Comment:
I don't think directory creation should have any significant performance impact. Also lazy creation just adds a bit of complexity in the code which I wanted to avoid. But I am open to this idea if you insist.